xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/XMLUtils.java

```diff
@@ -63,6 +64,59 @@
  */
 public final class XMLUtils
 {
+    /**
+     * A simple error logging helper to suppress expected error messages.
+     * While extracting text excerpts via the {@link #extractXML} method
+     * the extraction throws expected exceptions eg. in case a text node
+     * is too long. The default ErrorListener writes these errors to the console.
+     * However as these errors are expected under normal conditions
+     * and are either handled in the {@link #extractXML} method
+     * or passed up in the chain of commands it is ok to log these
+     * at a much lower level, where users normally never see them.
+     */
+    private static class RelaxedErrorListener implements ErrorListener
+    {
+        private static final String STACK_TRACE_NOTE = "stack trace for information only";
+
+        /**
+         * Fatal errors are unexpected and are logged as warnings here.
+         * They are not really fatal to XWiki but should be handled
+         * up the command chain later, or maybe propagated to the UI level.
+         */
+        @Override
+        public void fatalError(TransformerException exception) throws TransformerException
+        {
+            LOGGER.warn("fatal error from xml transformer", exception);
+        }
+
+        /**
+         * Errors are expected and are only logged as debug.
```

Comment:
Why are errors expected?

Reply:
Ok I think I understand now. You want to swallow "ERROR: 'Length limit reached'". I'm wondering if this is true for all errors, this makes them basically invisible and thus unknown.